tests/rptest/infinite_retention/infinite_retention_test.py

```diff
@@ -0,0 +1,794 @@
+            metrics_endpoint=MetricsEndpoint.PUBLIC_METRICS)
+        _stats["spillover_manifest_uploads_total"] = redpanda.metric_sum(
+            metric_name=
+            "redpanda_cloud_storage_spillover_manifest_uploads_total",
+            metrics_endpoint=MetricsEndpoint.PUBLIC_METRICS)
+
+        return _stats
+
+    def metadata_readable(self):
+        return next(self.rpk.describe_topic(
+            self.topic, tolerant=True)).high_watermark is not None
+
+    def _create_producer(self):
+        return KgoVerifierProducer(
+            self.test_context,
+            self.redpanda,
+            self.topic,
+            msg_size=self.params.msg_size,
+            msg_count=self.params.msg_count,
+            batch_max_bytes=512 * 1024,
+            rate_limit_bps=self.params.produce_byte_rate,
+            custom_node=[self.preallocated_nodes[0]],
+            debug_logs=True)
+
+    def _create_rnd_consumer(self):
+        return KgoVerifierRandomConsumer(self.test_context,
+                                         self.redpanda,
+                                         self.topic,
+                                         self.params.msg_size,
+                                         self.params.consumer_message_count,
+                                         self.params.parallel_consumers,
+                                         nodes=[self.preallocated_nodes[1]],
+                                         debug_logs=True,
+                                         trace_logs=True)
+
+    def _create_seq_consumer(self):
+        return KgoVerifierSeqConsumer(
+            self.test_context,
+            self.redpanda,
+            self.topic,
+            msg_size=self.params.msg_size,
+            # max_msgs=self.params.consumer_message_count,
+            max_msgs=None,  # consume everything
+            max_throughput_mb=self.params.consume_rate_mb *
+            self.params.parallel_consumers,
+            nodes=[self.preallocated_nodes[1]],
+            debug_logs=True,
+            trace_logs=True)
+
+    def _create_group_consumer(self):
+        return KgoVerifierConsumerGroupConsumer(
+            self.test_context,
+            self.redpanda,
+            self.topic,
+            msg_size=self.params.msg_size,
+            max_msgs=self.params.consumer_message_count,
+            readers=self.params.parallel_consumers,
+            max_throughput_mb=self.params.consume_rate_mb *
+            self.params.parallel_consumers,
+            nodes=[self.preallocated_nodes[1]],
+            debug_logs=True,
+            trace_logs=True)
+
+    @cluster(num_nodes=5)
+    @parametrize(restart_stress=True)
+    @parametrize(restart_stress=False)
+    def constant_throughput_prolonged_test(self, restart_stress):
+        """
+        Stress the tiered storage upload path on a multiple partitions
+        for a prolonged time. This ensures that there is no skips happen
```

Comment:
I would have guessed that "prolonged time" meant longer than the 5 minutes in this test.   Maybe "sustained over several minutes" would be  more accurate.